Enabled File plugin to handle shared memory areas.

karya0 opened the pull request:
The capability is removed from MTCP and moved completely into File plugin. I tested it with our test suite and everything seemed fine. It would be better if others can take it for a test ride.

For reviewing, I would suggest that you look at individual commits as some of the commits are somewhat unrelated.

This Closes #167, closes #156 and closes #69.

gc00 commented:
Hi @karya0.  Nice patch.  We've needed this for a while!  Here are some smaller comments:
1. Let's rebase  _Removed deprecated install-hooks api._ to be the first of the commits (or the last of the commits).  I expect that most of these commits will go into DMTCP 2.4.1, and We should do the deprecation for DMTCP 2.5, rather than for 2.4.1.  So, we don't want this commit in the middle of the others.
2.  See inline comment for _Better NSCD handling for writeckpt_, at: https://github.com/karya0/dmtcp/commit/91acdf9210f3c98001706592eb98f0b43d8f30ad.
3.  See inline comment for _Sys V SHM: Don't release mem region during ckpt._, at: https://github.com/karya0/dmtcp/commit/d53e389aa465ee0f07118dcd7a77eb9244d07dd2.
4. See inline comments for _writeckpt.cpp: Use ProcSelfMaps._, at: https://github.com/karya0/dmtcp/commit/26291ade3148b13e7e380cd43669ae907c441900.
5. And tiny inline comment for _	File: Moved msync from writeckpt.cpp to File plugin._, at: https://github.com/karya0/dmtcp/commit/ab735a9a2a16082fb773bf4aec35f9d978d24e73

karya0 commented:
I have addressed all the concerns. Here is the summary for the issues:

> Issue 1: install-hooks
It is the first commit, but somehow github UI is not showing it like that. However, if you track the commits through the parent link, you can see it. 

> Issue 2: mtcp_restart.c:fix_parent_directories()
This is already handled by the new code: https://github.com/dmtcp/dmtcp/pull/170/files#diff-e9f539d1f306bad2a4a3658de2364d59R305

gc00 commented:
It seems that @karya0 pushed this in, without the fix pointed out by @rohgarg above.  I've now added pull request #177, which fixes this, and also addresses a second issue that prevents us from building when configured with `--enable-debug`.  We can continue the discussion there.
